streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -56,32 +58,60 @@
                 final StateDirectory stateDirectory) {
         super(id, partitions, topology, consumer, true, stateMgr, stateDirectory, config);
 
-        closeTaskSensor = ThreadMetrics.closeTaskSensor(Thread.currentThread().getName(), metrics);
         processorContext = new StandbyContextImpl(id, config, stateMgr, metrics);
+        closeTaskSensor = ThreadMetrics.closeTaskSensor(Thread.currentThread().getName(), metrics);
+    }
+
+    @Override
+    public boolean isActive() {
+        return false;
     }
 
     @Override
-    public void initializeMetadata() {}
+    public State state() {
+        return state;
+    }
+
+    @Override
+    public void transitionTo(final State newState) {
```

Comment:
This logic is the same for standby / active, makes me thinking we can move the "state" field and this function to AbstractTask.